Ghidra/Features/WildcardAssembler/ghidra_scripts/FindInstructionWithWildcard.java

```diff
@@ -0,0 +1,346 @@
+
+		// 'allValidResolvedPatterns' has one entry for each encoding/wildcard value pair. We're
+		// going to reduce that down to a map where each:
+		// * Key is a single encoding of an instruction WITHOUT the wildcard operand
+		// bits specified
+		// * Value is a set of WildOperandInfo instances containing each valid wildcard
+		// completion
+		var encodings = new HashMap<AssemblyPatternBlock, HashSet<WildOperandInfo>>();
+		for (WildAssemblyResolvedPatterns x : allValidResolvedPatterns) {
+			var y = new ReducedWildcardAssemblyResolvedPattern(x);
+			var existing = encodings.get(y.maskedInstruction);
+			if (existing == null) {
+				existing = new HashSet<WildOperandInfo>();
+			}
+			existing.addAll(y.parent.getOperandInfo());
+			encodings.put(y.maskedInstruction, existing);
+		}
+		return encodings;
+	}
+
+	/**
+	 * This is a helper class which creates and holds an {@code AssemblyPatternBlock} for a given
+	 * {@code WildAssemblyResolvedPatterns}. This created {@code AssemblyPatternBlock} does NOT have
+	 * any bits specified that are part of a wildcarded operand. This is in contrast to the original
+	 * {@code WildAssemblyResolvedPatterns} where those bits are specified to have the values that
+	 * correspond to the {@code WildOperandInfo} values found in the
+	 * {@code WildAssemblyResolvedPatterns}.
+	 */
+	class ReducedWildcardAssemblyResolvedPattern {
+		/**
+		 * The original WildAssemblyResolvedPatterns that this is based on
+		 */
+		WildAssemblyResolvedPatterns parent;
+		/**
+		 * The portion of the instruction that is NOT any wildcarded operand
+		 */
+		AssemblyPatternBlock maskedInstruction;
+
+		ReducedWildcardAssemblyResolvedPattern(WildAssemblyResolvedPatterns input) {
+			parent = input;
+
+			// Remove all the bits which correspond to wildcarded opcodes from the instruction and
+			// save the result as maskedInstruction
+			var reducedInstruction = input.getInstruction();
+			for (WildOperandInfo info : input.getOperandInfo()) {
+				reducedInstruction = reducedInstruction.maskOut(info.location());
+			}
+			maskedInstruction = reducedInstruction;
+		}
+
+		/**
+		 * Returns true of the given value shares the same {@code maskedInstruction} and wildcard(s)
+		 * as this instance.
+		 * 
+		 * @param other
+		 *            Value to compare against
+		 * @return True if both values share the same maskedInstruction and wildcard(s)
+		 */
+		boolean sameBaseEncoding(ReducedWildcardAssemblyResolvedPattern other) {
+			if (this.maskedInstruction.compareTo(other.maskedInstruction) != 0) {
+				return false;
+			}
+
+			for (WildOperandInfo info : this.parent.getOperandInfo()) {
+				var foundMatch = false;
+				for (WildOperandInfo otherInfo : other.parent.getOperandInfo()) {
+					if (info.wildcard().equals(otherInfo.wildcard())) {
+						if (!info.equals(otherInfo)) {
+							return false;
+						}
+						foundMatch = true;
+						break;
+					}
+				}
+				if (!foundMatch) {
+					return false;
+				}
+			}
+			return true;
```

Comment:
I've reworked this a little bit and added comments. I'm fairly sure we can't use a map based on wildcard (name) because an instruction could have multiple wildcards in it with the same name.